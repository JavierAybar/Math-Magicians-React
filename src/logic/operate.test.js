import operate from './operate';

describe('correct function of operations', () => {
  test('addittion of two numbers', () => {
    expect(operate('1', '3', '+')).toMatchSnapshot();
  });

  test('substraction of two numbers', () => {
    expect(operate('3', '1', '-')).toMatchSnapshot();
  });

  test('multiplication of two numbers', () => {
    expect(operate('10', '2', 'x')).toMatchSnapshot();
  });

  test('division of two numbers', () => {
    expect(operate('12', '3', 'รท')).toMatchSnapshot();
  });

  test('division by zero', () => {
    expect(operate('2', '0', 'รท')).toMatchSnapshot();
  });

  test('modulo of two numbers', () => {
    expect(operate('10', '3', '%')).toMatchSnapshot();
  });

  test('modulo with divided by zero', () => {
    expect(operate('10', '0', '%')).toMatchSnapshot();
  });
});
