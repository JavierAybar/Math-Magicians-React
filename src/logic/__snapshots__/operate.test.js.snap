// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct function of operations addittion of two numbers 1`] = `"4"`;

exports[`correct function of operations division by zero 1`] = `"Can't divide by 0."`;

exports[`correct function of operations division of two numbers 1`] = `"4"`;

exports[`correct function of operations modulo of two numbers 1`] = `"1"`;

exports[`correct function of operations modulo with divided by zero 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`correct function of operations multiplication of two numbers 1`] = `"20"`;

exports[`correct function of operations substraction of two numbers 1`] = `"2"`;
